polygeist/tools/cgeist/Lib/clang-mlir.cc

```diff
@@ -4789,8 +4810,28 @@ mlir::Value MLIRASTConsumer::GetOrCreateGlobalLLVMString(
   return globalPtr;
 }
 
-mlir::func::FuncOp MLIRASTConsumer::GetOrCreateMLIRFunction(
-    const FunctionDecl *FD, const bool ShouldEmit, bool getDeviceStub) {
+llvm::Optional<mlir::FunctionOpInterface>
+MLIRASTConsumer::getFunction(const std::string &name,
+                             FunctionContext context) const {
+  const auto find = [&](const auto &map) {
+    const auto Iter = map.find(name);
+    return Iter == map.end()
+               ? llvm::Optional<mlir::FunctionOpInterface>{}
```

Comment:
When the function is not found we should return `llvm::None`

Reply:
This one's equivalent, but I guess `llvm::None` is more readable. Changed.